public/components/event_analytics/explorer/events_views/data_grid.tsx

```diff
@@ -114,8 +123,12 @@ export function DataGrid(props: DataGridProps) {
   const dataGridColumns = () => {
     const columns: EuiDataGridColumn[] = [];
     selectedColumns.map(({ name }) => {
-      if (name === 'timestamp') {
-        columns.push(DEFAULT_TIMESTAMP_COLUMN);
+      if (name === defaultTimestamp) {
```

Comment:
This seems to be a little bit confusing to me. The map is supposed to help you create a new mapped columns just like you defined in the first place, but later you are doing a push in the map.

Please consider either use map or forEach. Seems like map here should be good for your case.

Reply:
I am not fully understanding your confusion here, in this method I have selectedColumns which I essentially reformat into an array of `EuiDataGridColumn`, for which I'm pushing into. Could you clarify what's wrong here, is it with the timestamp case specifically?